cmd/imagescan/collector/collector.go

```diff
@@ -69,6 +69,9 @@ func (c *Collector) Collect(ctx context.Context) error {
 	}
 	defer cleanup()
 
+	ctx, cancel := context.WithTimeout(ctx, c.cfg.Timeout)
```

Comment:
Ok so all in all we now apply the timeout after we fetched the image from the registry. 

I am not sure if I like this, as this could potentially also be stuck forever mhm.

Reply:
I haven't thought about that possibility but I agree, moved it to the top of the function